### ISSUE TITLE
InvalidCastException at UnityEngine.UnitySynchronizationContext.ExecuteTasks

### ISSUE THREAD

**Issue by zed9h** [github-for-unity/Unity#76]
### Description

After installing the Github plugin v0.12-alpha (b9289f4) on Unity 2017.1.0f1 it requests an update

![screenshot 2017-07-02 12 43 53](https://user-images.githubusercontent.com/7734/27771490-c13da41c-5f25-11e7-8b6c-bdb7dad4f55e.png)

(accepting or denying it have the same effect)
and the editor log finishes like that

```
Initializing GitHub for Unity version 0.12.0.0
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
GitHub.Unity.EntryPoint:Initialize() (at /Users/shana/code/work/github/Unity/src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs:86)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions() (at C:\buildslave\unity\build\Editor\Mono\EditorApplication.cs:109)
 
(Filename: /Users/shana/code/work/github/Unity/src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs Line: 86)

Initializing GitHub for Unity log file: C:\Users\zed\AppData\Local\GitHubUnity\github-unity.log
UnityEngine.DebugLogHandler:Internal_Log(LogType, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
GitHub.Unity.EntryPoint:Initialize() (at /Users/shana/code/work/github/Unity/src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs:86)
UnityEditor.EditorApplication:Internal_CallUpdateFunctions() (at C:\buildslave\unity\build\Editor\Mono\EditorApplication.cs:109)
 
(Filename: /Users/shana/code/work/github/Unity/src/UnityExtension/Assets/Editor/GitHub.Unity/EntryPoint.cs Line: 86)
```

with this message repeated multiple times

```
InvalidCastException: Cannot cast from source type to destination type.
  at UnityEngine.UnitySynchronizationContext.ExecuteTasks () [0x0000b] in C:\buildslave\unity\build\Runtime\Export\UnitySynchronizationContext.cs:56 
 
(Filename: C:/buildslave/unity/build/Runtime/Export/UnitySynchronizationContext.cs Line: 56)
```

this message also repeats a lot on every play.

### Steps to Reproduce

1. Install Unity 2017.1.0f1
2. Install the Github plugin v0.12-alpha (b9289f4)
3. Look at the console or the Editor.log

**Expected behavior:** No errors to be printed

**Actual behavior:** Lots of (compilation?) errors

**Reproduces how often:** every time

### Additional Information

Not sure if it's a Unity's problem, but it is clearly triggered by the Github plugin.

**Comment by StanleyGoldman** [comment]
Hey @zed9h thanks for trying and it's definitely our problem, but right now we are only looking to support unity 5.4, 5.5 and 5.6 (and a pull request in the works for 5.3).

We will probably wait until we are out of alpha/beta, before we look to support 2017.1

**Comment by zed9h** [comment]
@StanleyGoldman it makes sense. I posted it now because 2017.1.0f1 is final (or release candidate), and probably will replace the 5.6 as stable in the next few days (as it is scheduled on the roadmap for July 2017).

**Comment by StanleyGoldman** [comment]
I didn't realize it was in release candidate yet. 😨 
Thanks for the info.

**Comment by jwvanderbeck** [comment]
Arg.  I need to read readme's better, but just ran into this myself.  With Unity pushing 2017.1 and this project "coming out" so to speak I just made the sutpidly incorrect assumption it would be good, but nope.

Do you know what is causing this? Maybe something we can relatively easily look into and fix on our end?

I'm really looking for a Collab alternative right now because it has cost me so much lost time with its instability.  Currently using GitHub and GitLFS standard without the plugin, but while that is fine for me and code, it isn't artist friendly.

**Comment by StanleyGoldman** [comment]
Fixed with #175
